src/mame/drivers/didact.cpp

```diff
@@ -457,34 +451,381 @@ void mp68a_state::machine_start()
+{
+	int x, y;
+	int vramad;
+	UINT8 *chardata;
+	UINT8 charcode;
+
+	LOGSCREEN(("%s()\n", FUNCNAME));
+	vramad = 0;
+	for (int row = 0; row < 32 * 8; row += 8)
+	{
+		for (int col = 0; col < 32 * 8; col += 8)
+		{
+			/* look up the character data */
+			charcode = m_vram[vramad];
+			if (VERBOSE && charcode != 0x20 && charcode != 0) LOGSCREEN(("\n %c at X=%d Y=%d: ", charcode, col, row));
+			chardata = &m_char_ptr[(charcode * 8)];
+			/* plot the character */
+			for (y = 0; y < 8; y++)
+			{
+				if (VERBOSE && charcode != 0x20 && charcode != 0) LOGSCREEN(("\n  %02x: ", *chardata));
+				for (x = 0; x < 8; x++)
+				{
+					if (VERBOSE && charcode != 0x20 && charcode != 0) LOGSCREEN((" %02x: ", *chardata));
+					bitmap.pix16(row + y, col + x) = (*chardata & (1 << x)) ? 1 : 0;
+				}
+				chardata++;
+			}
+			vramad++;
+		}
+		if (VERBOSE && charcode != 0x20 && charcode != 0) LOGSCREEN(("\n"));
+	}
+
+	return 0;
+}
+
+/* PIA write - the Esselte 100 allows the PIA:s to be accessed simultaneously */
+WRITE8_MEMBER( e100_state::pia_w )
+{
+	LOG(("%s(%02x)\n", FUNCNAME, data));
+	if ((offset & 0x08) == 0x08)
+	{
+		LOG(("- PIA1\n"));
+		m_pia1->write(space, offset, data);
+	}
+	if ((offset & 0x10) == 0x10)
+	{
+		LOG(("- PIA2\n"));
+		m_pia2->write(space, offset, data);
+	}
+	if (VERBOSE && (offset & 0x18) == 0x18)
+	{
+		LOGCS(("- Dual device write access!\n"));
+	}
+	if (VERBOSE && (offset & 0x18) == 0x00)
+	{
+		logerror("- Funny write at offset %02x!\n", offset);
+	}
+}
+
+/* PIA read  - the Esselte 100 allows the PIA:s to be accessed simultaneously */
+READ8_MEMBER( e100_state::pia_r )
+{
+	UINT8 data = 0;
+
+	switch (offset & 0x18)
+	{
+	case 0x18: // read PIA1 and PIA2 at the same time, should really only happen for writes...
+		{
+			int data1 =  m_pia1->read(space, offset);
+			int data2 =  m_pia2->read(space, offset);
+			logerror("%s: Dual device read may have caused unpredictable results on real hardware\n", FUNCNAME);
+			data = data1 & data2; // We assume that the stable behaviour is that data lines with a low level by either device succeeds
+			LOGCS(("%s %s[%02x] %02x & %02x -> %02x Dual device read!!\n", PIA1_TAG "/" PIA2_TAG, FUNCNAME, offset, data1, data2, data));
+		}
+		break;
+	case 0x08: // PIA1
+		data = m_pia1->read(space, offset);
+		LOGCS(("%s %s(%02x)\n", PIA1_TAG, FUNCNAME, data));
+		break;
+	case 0x10: // PIA2
+		data = m_pia2->read(space, offset);
+		LOGCS(("%s %s(%02x)\n", PIA2_TAG, FUNCNAME, data));
+		break;
+	default: // None of the devices are selected
+		logerror("%s: Funny read at offset %02x\n", FUNCNAME, offset);
+	}
+	return data;
+}
+
+WRITE8_MEMBER( e100_state::pia1_kbA_w )
+{
+	LOG(("%s(%02x)\n", FUNCNAME, data));
+}
+
+READ8_MEMBER( e100_state::pia1_kbA_r )
+{
+	int ls145;
+	UINT8 pa = 0x00;
+
+	// Read out the selected column
+	ls145 = m_kbd_74145->read() & 0x3ff;
+
+	// read out the artwork
+	switch (ls145)
+	{
+	case 0: pa = 0x00; break;
```

Comment:
If this is a typical matrix keyboard, driving multiple rows simultaneously will result in all the associated rows being wired-anded together.  Please implement this (see e.g. osborne1 as an example) or show evidence that there's some lockout causing different behaviour when multiple rows are driven.